src/proxy/outbound.rs

```diff
@@ -136,6 +159,33 @@ pub struct OutboundConnection {
     pub hbone_port: u16,
     pub metrics: Arc<Metrics>,
     pub id: TraceParent,
+    pub pool: Pool<PoolValue, PoolKey>,
+}
+
+#[derive(Debug, Clone)]
+pub struct PoolClient(http2::SendRequest<Body>);
+
+impl Poolable for PoolClient {
+    fn is_open(&self) -> bool {
+        // TODO
+        true
+    }
+
+    fn reserve(self) -> Reservation<Self> {
+        let b = self.clone();
+        let a = self;
+        Reservation::Shared(a, b)
+    }
+
+    fn can_share(&self) -> bool {
+        true // http2 always shares
+    }
+}
+
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub struct PoolKey {
+    src: Identity,
+    dst: Identity,
```

Comment:
With src & dst identity, it could not perfectly reflect the relationship of connections number and the workload numbers. 

It is very common to have multi pods within a node with same identity

Reply:
The original intent was to explicitly pool those. What is the motivation to not pool them?

Reply:
This is not a big problem as there is no connection pool size limit, it is max_idle_per_host.

Reply:
wondering how to map the pods with same identity to the same connection? In the HTTP2 stream?

Reply:
Yes, they are multiplexed. Each user connection is a H2 stream

Reply:
Thank you, John, I think this pool would significantly improve the performance, especially for some benchmarks, looking forward to it.